### PR TITLE
ES6-ify and fix lint

### PR DESCRIPTION
**Pull request by joefitter**
* Removed underscore, replaced with lodash
* installed eslint-config-homeoffice
* fixed linting errors

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,12 +4,12 @@
...
   "scripts": {
-    "test": "npm run unit && npm run cover && npm run snyk",
+    "test": "npm run lint && npm run unit && npm run cover && npm run check-coverage && npm run snyk",
     "unit": "mocha --recursive test/spec/ --require test/helper --timeout 300000",
     "cover": "istanbul cover _mocha -- -R dot --recursive test/spec/ --require test/helper --timeout 300000",
-    "check-coverage": "istanbul check-coverage --statement 98.31 --branch 94.89 --function 96.30 --line 98.31",
+    "check-coverage": "istanbul check-coverage --statement 90 --branch 90 --function 90 --line 90",
     "lint": "eslint .",
-    "snyk": "snyk test",
+    "snyk": "snyk auth $SNYK_TOKEN && snyk test",
```
Comment (lennym) [inline package.json]: This needs to be `$SNYK_AUTH` not `$SNYK_TOKEN` judging by the current travis settings.

screen shot 2017-02-10 at 17 32 19

**Review by lennym — CHANGES_REQUESTED** [review]

File: lib/validation/index.js [hunk 2]
```diff
@@ -1,104 +1,106 @@
...
-        if (!_.isArray(validator.arguments)) {
-            validator.arguments = [ validator.arguments ];
-        }
-        args = args.concat(validator.arguments);
-        if (!validators[validator.type].apply(null, args)) {
-            return _.extend({ key: key }, validator, { arguments: validator.arguments });
-        }
+  function doValidate() {
+    debug('Applying %s validator with value "%s"', validator.type, value);
+    var args = [value];
```
Comment (JoeChapman) [inline lib/validation/index.js]: if es6-ifying, why not use `let` here?

File: package.json [hunk 3]
```diff
@@ -4,12 +4,12 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "npm run unit && npm run cover && npm run snyk",
+    "test": "npm run lint && npm run unit && npm run cover && npm run check-coverage && npm run snyk",
     "unit": "mocha --recursive test/spec/ --require test/helper --timeout 300000",
     "cover": "istanbul cover _mocha -- -R dot --recursive test/spec/ --require test/helper --timeout 300000",
-    "check-coverage": "istanbul check-coverage --statement 98.31 --branch 94.89 --function 96.30 --line 98.31",
+    "check-coverage": "istanbul check-coverage --statement 90 --branch 90 --function 90 --line 90",
```
Comment (JoeChapman) [inline package.json:10]: why 90?

File: package.json [hunk 4]
```diff
@@ -4,12 +4,12 @@
...
     "lint": "eslint .",
-    "snyk": "snyk test",
+    "snyk": "snyk auth $SNYK_AUTH && snyk test",
```
Comment (JoeChapman) [inline package.json:12]: oh cool so we set the env var `SNYK_AUTH`. Is it only set during CI?

File: package.json [hunk 5]
```diff
@@ -29,17 +29,17 @@
     "hof-behaviour-session": "^1.0.0",
     "hogan.js": "^3.0.2",
     "i18n-lookup": "^0.1.0",
-    "lodash.clonedeep": "^4.5.0",
+    "lodash": "^4.17.4",
     "mixwith": "^0.1.1",
     "moment": "^2.9.0",
```
Comment (JoeChapman) [inline package.json:34]: Perhaps use the latest; `2.17.1 `

Comment (JoeChapman) [reply package.json:34]: appreciate this is not an attempt to update all deps though

File: package.json (hunk 3 above)
Comment (joefitter) [reply package.json:10]: why 98.31?

File: package.json (hunk 4 above)
Comment (joefitter) [reply package.json:12]: yeah - in travis config using travis encrypt

File: package.json (hunk 5 above)
Comment (joefitter) [reply package.json:34]: this will still get 2.17.1

File: lib/validation/index.js (hunk 2 above)
Comment (joefitter) [reply lib/validation/index.js]: missed one, updating!

File: package.json (hunk 3 above)
Comment (JoeChapman) [reply package.json:10]: precisely

Comment (JoeChapman) [reply package.json:10]: 0 or 100

Comment (joefitter) [reply package.json:10]: it was 98.31 as that's what hmpo set it to, as that was their current coverage, and they wanted to make sure it didn't drop when new code was added. Which is fair. _however_ adding all the code and tests from hof-base-controller results in test coverage dropping by like half a percent. And seeing as this isn't a PR to increase test coverage I set it to 90 to get the build to pass. I could just have removed the check, happy to do this if you prefer?

File: package.json (hunk 5 above)
Comment (JoeChapman) [reply package.json:34]: uh, yep sorry.

File: package.json (hunk 3 above)
Comment (JoeChapman) [reply package.json:10]: It doesn't hut to have it there. (AFAIK) I just thought it was an arbitrary choice.

File: package.json (hunk 4 above)
Comment (JoeChapman) [reply package.json:12]: that was my next question (encryption) nice one.

File: package.json (hunk 3 above)
Comment (joefitter) [reply package.json:10]: it is, kinda... more than 0, less than 100

**Review by lennym — APPROVED** [review]

**Pull request merged**